river/river.go

```diff
@@ -232,16 +232,23 @@ func (r *River) prepareRule() error {
 		}
 	}
 
-	for _, rule := range r.rules {
+	rules := make(map[string]*Rule)
+	for key, rule := range r.rules {
 		if rule.TableInfo, err = r.canal.GetTable(rule.Schema, rule.Table); err != nil {
 			return errors.Trace(err)
 		}
 
 		if len(rule.TableInfo.PKColumns) == 0 {
-			return errors.Errorf("%s.%s must have a PK for a column", rule.Schema, rule.Table)
+			if r.c.SkipNonPk == false {
+				return errors.Errorf("%s.%s must have a PK for a column", rule.Schema, rule.Table)
+			} else {
+				fmt.Printf("ignored table without primary key: %s\n", rule.TableInfo.Name)
```

Comment:
use log.Errorf

Reply:
errors.Errorf("%s.%s must have a PK for a column", rule.Schema, rule.Table) to log.Errorf...?
or fmt.Printf("ignored table without primary key: %s\n", rule.TableInfo.Name) to log.Errorf...?

Reply:
fmt.Printf -> log.Errorf